packages/serverpod/lib/src/server/session.dart

```diff
@@ -257,8 +257,17 @@ class MethodCallSession extends Session {
     queryParameters.addAll(uri.queryParameters);
     this.queryParameters = queryParameters;
 
-    var methodName = queryParameters['method'];
-    if (methodName == null && endpointName == 'webserver') methodName = '';
+    String? methodName;
+    if (path.contains('/')) {
+      var pathComponents = path.split('/');
+      endpointName = pathComponents[0];
+      methodName = pathComponents[1];
+    } else {
+      endpointName = path;
+    }
+
+    methodName ??= queryParameters['method'];
+    if (methodName == null && path == 'webserver') methodName = '';
```

Comment:
Suggestion: Move to else statement.

If I understand this correctly, this logic is only applicable in the case that path does not contain `'/'`, otherwise we set method name to `pathComponens[1]`, and that will never set `methodName` to `null`.

Reply:
This means you could probably remove the nullability of the `methodName` variable and therefore get rid of the `!` below.

Reply:
Also seems like the current logic could let `null` through if `queryParameters['method']` is null but `path` is not `'webserver'`.

Reply:
Fixed